explorer/testdata/assoc_const/fail_multi_impl_scoping.carbon

```diff
@@ -28,17 +28,16 @@ class C(T:! Type) {
     fn FB() -> i32 {
       // OK, know that TB is i32 here.
       let v: Self.(B.TB) = 2;
-      // TODO: Don't know that TA is i32. It could be specialized.
-      // We should reject this once we support specialization.
+      // Don't know that TA is i32; it could be specialized.
       let w: Self.(A.TA) = 3;
+      // CHECK:COMPILATION ERROR: {{.*}}/explorer/testdata/assoc_const/fail_multi_impl_scoping.carbon:[[@LINE+1]]: type error in return value: '((class C(T = T)).TB).Result' is not implicitly convertible to 'i32'
```

Comment:
This error feels weird, particularly if line 29 says it's known to be i32

Reply:
Also I'm a little confused where `.Result` comes from

Reply:
I think what this means is that `(C(T).TB).(AddWith(C(T).TA).Result)` is not implicitly convertible to `i32`, but our value printing doesn't print out the `AddWith(C(T).TA).` part.

I've added back a TODO; I think we're still not handling this properly (and there's no reason why this change would have fixed it -- I think it may have only changed the diagnostic because it makes the `impl` below not visible here).